api/maven-api-settings/src/main/mdo/settings.mdo

```diff
@@ -535,6 +539,62 @@
           <type>String</type>
         </field>
       </fields>
+      <codeSegments>
+        <codeSegment>
+          <version>1.0.0/1.3.0</version>
+          <code>
+            <![CDATA[
+    public boolean isActive() {
+        return (getActiveString() != null) ? Boolean.parseBoolean(getActiveString()) : true;
+    }
+
+    public void setActive(boolean active) {
+        setActiveString(String.valueOf(active));
+    }
+
+    public int getPort() {
+        return (getPortString() != null) ? Integer.parseInt(getPortString()) : 8080;
+    }
+
+    public void setPort(int port) {
+        setPortString(String.valueOf(port));
+    }
+            ]]>
+          </code>
+        </codeSegment>
+        <codeSegment>
+          <version>2.0.0+</version>
+          <code>
+            <![CDATA[
+    /**
+     * Indicates if this proxy is active or not.
+     * To allow interpolation of this field, this method lazily parses
+     * the {@link #getActiveString()} value as a boolean and defaults to {@code true}
+     * if not set.
+     *
+     * @return a boolean indicating whether this proxy is active or not
+     */
+    public boolean isActive() {
+        return (getActiveString() != null) ? Boolean.parseBoolean(getActiveString()) : true;
+    }
+
+    /**
+     * Returns the port to use for this proxy.
+     * To allow interpolation of this field, this method lazily parses
+     * the {@link #getPortString()} value as an integer and defaults to {@code 8080}
+     * if not set.
+     *
+     * @return an integer indicating the port to use for this proxy
+     * @throws NumberFormatException if the port string is not a valid integer
```

Comment:
This should happen when the port string is set, not later when the getter method is called

Reply:
This is usually not set manually, but only when parsing the xml and I've only added the exception because it is actually thrown.  Also, the whole idea is to allow interpolation, so I can't restrict `setPortString` to valid integers, else, the whole PR is pointless.

Reply:
NumberFormatException is an IllegalArgumentException. It is very weird for a method with no arguments to throw an IllegalArgumentException and very unexpected for a getter to throw any exception at all.

Where/when does interpolation happen? Can you throw the exception at that point and store an int for later use here?

Reply:
Fwiw, the port can be negative, in which case a default value will be used, depending on the protocol.  At least, that what is done in the resolver for the http transport.  So I'll abstain from any semantic validation, but I'll add some code to ensure that it can be parsed as an integer.